### PR TITLE
Fix #82 - GET collection with input filters

### LINKED ISSUE #82: Rest content validation issue
In rest service (fetch all for example), if I seet **allows_only_fields_in_filter** for **GET** method
I will got this kind of error `Unrecognized fields: 0`
This because content something like this
```php
array (size=1)
  0 => 
    array (size=1)
      'server_name' => string 'asdasd'
```

if it is not a get method the content of values are this
```php
  array (size=1)
      'server_name' => string 'asdasd'
```

This problem just in REST fetch all, please have a look and tell me how I can fix this for now, before your update.

Thanks.

### PR DESCRIPTION
**Pull request by webimpress**
Please see discussion in #82.

Tests could be done better. Just wanted to finish it ...
What do you think about dataProvider?

### CODE REVIEW

File: test/ContentValidationListenerTest.php [hunk 1]
```diff
@@ -129,7 +174,7 @@ public function testReturnsApiProblemResponseIfCollectionRequestWithoutBodyIsInv
         ]));
         $listener = new ContentValidationListener([
             'Foo' => ['GET' => 'FooValidator'],
-        ], $services);
+        ], $services, ['Foo' => 'foo_id']);
```
Comment (weierophinney) [inline test/ContentValidationListenerTest.php]: Why was this change necessary?

Comment (webimpress) [inline test/ContentValidationListenerTest.php]: Because without this it was entity request. Please check method `isCollection`:
https://github.com/zfcampus/zf-content-validation/blob/master/src/ContentValidationListener.php#L430-L432

Comment (weierophinney) [inline test/ContentValidationListenerTest.php]: Cool - so, essentially the test was not correctly written previously. With the additional tests you wrote, this becomes clearer. Thanks!
